w09h03/test/pgdp/filter/DGTest.java

```diff
@@ -0,0 +1,5539 @@
+        int number = frame.getFrameNumber();
+
+        String output = Operations.decode(frame);
+
+        Assertions.assertEquals("e", output);
+
+        // test frame nicht verändert
+        Assertions.assertEquals(width, frame.getWidth());
+        Assertions.assertEquals(height, frame.getHeight());
+        Assertions.assertEquals(number, frame.getFrameNumber());
+        for(int i = 0; i < 8; i++){
+            Assertions.assertEquals(colorsE[i], frame.getPixels().getRGB(i, frame.getHeight()-1));
+        }
+        for(int i = 8; i < 14; i++){
+            Assertions.assertEquals(colorsE[i-8], frame.getPixels().getRGB(i, frame.getHeight()-1));
+        }
+        for(int i = 16; i < frame.getWidth(); i++){
+            Assertions.assertEquals(Color.YELLOW.getRGB(), frame.getPixels().getRGB(i, frame.getHeight()-1));
+        }
+        for(int i = 0; i < frame.getWidth(); i++){
+            for(int j = 0; j < frame.getHeight()-1; j++){
+                Assertions.assertEquals(Color.YELLOW.getRGB(), frame.getPixels().getRGB(i, j));
+            }
+        }
+
+    }
+
+    @Test // test für leeres image nicht nötig, da konstruktor von bufferedimage exception wirft bei width, height <= 0
+    @DisplayName("Test DG 09 Decode null")
+    public void test09(){
+
+        try{
+            String output = Operations.decode(null);
+        }
+        catch(Exception e){
+            Assertions.fail();
+        }
+
+    }
+
+    // OPERATIONS ENCODE FRAME TESTS
+
+    @Test
+    @DisplayName("Test DG 20 Encode neues BufferedImage 500x500 mit Buchstabe e")
+    public void test20() throws IOException {
+        int size = 500;
+        BufferedImage image = new BufferedImage(size,size,BufferedImage.TYPE_INT_ARGB);
+        for(int i = 0; i < size; i++){
+            for(int j = 0; j < size; j++){
+                image.setRGB(i,j,Color.YELLOW.getRGB());
+            }
+        }
+        Frame frame = new Frame(image, 0);
+
+        int height = frame.getHeight();
+        int width = frame.getWidth();
+        int number = frame.getFrameNumber();
+
+        Function<Frame, Frame> function = Operations.encode("e");
+        //Frame output = frame;
+        Frame output = function.apply(frame);
+        System.out.println(Color.RED.getRGB());
+        //System.out.println(Color.BLACK.getRGB());
+        //System.out.println(Color.WHITE.getRGB());
+        int[] colorsE = new int[]{Color.BLACK.getRGB(), Color.WHITE.getRGB(), Color.WHITE.getRGB(),
+                Color.BLACK.getRGB(), Color.BLACK.getRGB(), Color.WHITE.getRGB(), Color.BLACK.getRGB(), Color.WHITE.getRGB()};
+
+        Assertions.assertEquals(width, output.getWidth());
+        Assertions.assertEquals(height, output.getHeight());
+        Assertions.assertEquals(number, output.getFrameNumber());
+        // encoding path
+        for(int i = 0; i < 8; i++){
+            //System.out.println("expected:"+nums[i]+", acutally:"+output.getPixels().getRGB(i, output.getHeight()-1));
```

Comment:
Code that is commented out should not belong into a PR

Reply:
True, see above.